src/context/SidebarContext.tsx

```diff
@@ -0,0 +1,27 @@
+'use client';
+
+import {createContext, ReactNode, useState} from 'react';
+
+type SidebarContextProps = {
+	expanded: boolean;
+	setExpanded: (value: boolean) => void;
+};
+
+export const SidebarContext = createContext<SidebarContextProps>({
+	expanded: true,
+	setExpanded: () => {},
+});
+
+interface FontProviderProps {
+	children: ReactNode;
+}
+
+export const SidebarProvider: React.FC<FontProviderProps> = ({children}) => {
+	const [expanded, setExpanded] = useState<boolean>(true);
```

Comment:
Tu n'as pas besoin de mettre un type à ton useState si tu lui donne une valeur précise, comme lorsque tu déclare une variable ;) 

```suggestion
	const [expanded, setExpanded] = useState(true);
```